Play nice with other processes.

FurryFuttock opened the pull request:
Add a time wait for packets, this reduces CPU usage when idle, but doesn't affect throughput.

The timeout is in seconds as that is fine for my usage, I suppose that there is a case for having it in milliseconds or microseconds.

FurryFuttock committed:
Add a time wait for packets, this reduces CPU usage when idle, but doesn't affect throughput.

FurryFuttock committed:
Now builds on Windows under Visual Studio 2022

We still have lots of warnings.

FurryFuttock committed:
Don't warn on scanf when using Visual Studio

liftoff-sr reviewed (commented):
I'll accept the patch if modified to use microseconds.  Then simply add 6 zeros to your current value to obtain seconds of delay.  I like that it defaults to zero.

FurryFuttock committed:
Windows doesn't let us bind to INADDR_BROADCAST, so we build the interface specific broadcast address.

FurryFuttock committed:
Windows doesn't let us bind to the interface specific broadcast address, so bind to INADDR_ANY.

FurryFuttock committed:
Finally got this to work on Windows under Visual Studio.

Winsock2 only seems seem to want bind broadcast UDP sockets to INADDR_ANY, so for _WIN32 builds I bind udp_global_broadcast_listener and disable  udp_local_broadcast_listener.

triomorp committed:
Now builds under gcc 12

FurryFuttock committed:
CIPster classes return size as int not size_t, so we must use format %d, not %zd

FurryFuttock committed:
Current versions of CMake don't support cmake_minimum_required less than 3.5

FurryFuttock committed:
Add missing cstdint includes

liftoff-sr reviewed (commented):
Your comment is not true.  The return type is ssize_t, not int.  ssize_t is posix compatible but not part of the C++ standard.  If your compiler is having problems with ssize_t then we should consider using ptrdiff_t as the return type of ByteBuf size() functions and stay with %zd.

liftoff-sr commented on source/src/byte_bufs.h:
approved

liftoff-sr commented on source/src/enet_encap/sockaddr.cc:
approved

liftoff-sr commented on source/CMakeLists.txt:
approved

liftoff-sr commented on source/src/cip/cipepath.h:
approved

liftoff-sr commented:
Please submit new merge requests for future changes.  The chances of a full merge request going in diminish with size.
